### PR TITLE
ImageLocationScanner: Use core 2.10

### PR DESCRIPTION
**Pull request by kingthorin**
- Update build file, and CHANGELOG.
- ImageLocationScanRule > Upgrade to more recent Alert and parent handling, and leverage newer logging infrastructure.

Part of: zaproxy/zaproxy#6376

Signed-off-by: kingthorin <kingthorin@users.noreply.github.com>

### CODE REVIEW

**Comment by kingthorin** [comment]
I decided it would be nice to tackle a smaller one to get the testutils changes in and ensure we were all on the same page.

I'll rebase the others once this is merged. Also, I'll tackle spotlessApply on the ImageLocationScanRule in another PR.

**Comment by kingthorin** [comment]
Will investigate the test failure.

**Comment by thc202** [comment]
The testutils could be updated in its own PR.

**Comment by thc202** [comment]
Re the test failure, you need to change the tests to call `scanHttpResponseReceive(HttpMessage)` instead.

File: addOns/imagelocationscanner/CHANGELOG.md [hunk 1]
```diff
@@ -4,7 +4,8 @@ All notable changes to this add-on will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 
 ## Unreleased
-
+### Changed
+- Now using 2.10 logging infrastructure (Log4j 2.x).
```
Comment (thc202) [inline addOns/imagelocationscanner/CHANGELOG.md:8]: Say that the minimum ZAP version is now 2.10 instead?

File: addOns/imagelocationscanner/imagelocationscanner.gradle.kts [hunk 2]
```diff
@@ -6,11 +6,12 @@ description = "Image Location and Privacy Passive Scanner"
 zapAddOn {
     addOnName.set("Image Location and Privacy Scanner")
     addOnStatus.set(AddOnStatus.BETA)
-    zapVersion.set("2.9.0")
+    zapVersion.set("2.10.0")
 
     manifest {
         author.set("Jay Ball (veggiespam) and the ZAP Dev Team")
         url.set("https://www.zaproxy.org/docs/desktop/addons/image-location-and-privacy-scanner/")
+        notBeforeVersion.set("2.10.0")
```
Comment (thc202) [inline addOns/imagelocationscanner/imagelocationscanner.gradle.kts:14]: This is not necessary (already implied with `zapVersion` above).

**Comment by thc202** [comment]
Re the commit message, the changelog should be singular and the issue references could be tweaked like done in the issue.

**Comment by kingthorin** [comment]
https://github.com/zaproxy/zap-extensions/pull/2678

**Comment by kingthorin** [comment]
All fixed.

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Review by psiinon — APPROVED** [review]

**Comment by kingthorin** [comment]
Thanks!

**Pull request merged**